Marlin/src/feature/bedlevel/bedlevel.cpp

```diff
@@ -154,7 +154,7 @@ void reset_bed_level() {
       #endif
       LOOP_L_N(x, sx) {
         SERIAL_CHAR(' ');
-        const float offset = values[x * sx + y];
+        const float offset = values[x * sy + y];
```

Comment:
If anything, I would guess `[(y * sx + x]` here (taking `x` as columns, `y` as rows). But, it depends on what the dataset order is expected. I'll check this all out in the simulator with data set by `M421`, etc.

Reply:
Aha, yes. I see `bed_mesh_t` is "`sy`" wide and "`sx`" tall. Patching everywhere, thanks!

Reply:
> Aha, yes. I see `bed_mesh_t` is "`sy`" wide and "`sx`" tall. Patching everywhere, thanks!

It took me quite some time to come to the same conclusion. The original bug was mine so I guess you could say you saw some of the evolution of my thinking.